rholang/src/main/scala/coop/rchain/rholang/interpreter/matcher/SpatialMatcher.scala

```diff
@@ -50,17 +50,12 @@ object SpatialMatcher extends SpatialMatcherInstances {
 
     val doMatch: R[Unit] = SpatialMatcher.spatialMatch[R, Par, Par](target, pattern)
 
-    val matchAndCharge: M[Option[(FreeMap, Unit)]] = for {
-      phlosAvailable <- cost.get
-      result <- runFirstWithCost[M, Unit](doMatch).onError {
+    val matchAndCharge: M[Option[(FreeMap, Unit)]] =
+      runFirstWithCost[M, Unit](doMatch).onError {
                  case OutOfPhlogistonsError =>
                    // if we run out of phlos during the match we have to zero phlos available
                    cost.get.flatMap(ca => charge[M](ca))
                }
```

Comment:
I don't think we need the `onError` anymore, as the OOPE error will be due to no phlo left in the now-shared cost account.

Reply:
BTW: this method starts to finally look like God intended it to be ;)